pilot/pkg/serviceregistry/serviceentry/conversion.go

```diff
@@ -43,6 +43,25 @@ func convertPort(port *networking.Port) *model.Port {
 	}
 }
 
+func getTargetPortFromServiceInstances(port *model.Port, svc *model.Service, serviceInstances []*model.ServiceInstance) uint32 {
+	if port == nil {
+		return uint32(0)
+	}
+	if len(serviceInstances) == 0 {
+		return uint32(0)
+	}
+
+	for _, instance := range serviceInstances {
+		if svc != instance.Service {
+			continue
+		}
+		if port.Name == instance.ServicePort.Name {
+			return instance.Endpoint.EndpointPort
```

Comment:
There may be 100 different "target ports" for one service. Trying to encode a single target port for a service is not consistent with Kubernetes Service

Reply:
How about to iterate them one by one for the Service Entry, and append these ports into the Service Entry's Ports? Any idea?

Reply:
Is there any problem for a Service Entry has multiple ports with the same port name but different target names?

Reply:
I think the function is fundamentally broken and cannot be fixed... a ServiceEntry cannot represent a Service properly

Reply:
@howardjohn @zhlsunshine It seems that there's no perfect solution for this issue. How about we do it this way:
If the k8s svc target port is a number, we put that number at the target port of the service entry, otherwise, leave it as it is now.  This at least would partly solve this problem. And I can ask the users to change the target port name to a number if they encounter this issue. 99% of svcs instances that I know have the same target number.  So this would work on most occasions.